src/us/kbase/workspace/database/provenance/Organization.java

```diff
@@ -0,0 +1,92 @@
+package us.kbase.workspace.database.provenance;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Objects;
+import java.util.Optional;
+
+public class Organization {
+	private final String organizationName;
+	private final String organizationID;
+
+	private Organization(final String organizationName, final String organizationID) {
+		this.organizationName = organizationName;
+		this.organizationID = organizationID;
+	}
+
+	/**
+	 * Get the name of the organization, for example KBase.
+	 *
+	 * @return the organization name.
+	 */
+	public String getOrganizationName() {
+		return organizationName;
+	}
+
+	/**
+	 * Get the ID of the organization, for example https://ror.org/03rmrcq20.
+	 *
+	 * @return the data id, if present.
+	 */
+	public Optional<String> getOrganizationID() {
+		return Optional.ofNullable(organizationID);
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(organizationID, organizationName);
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if ((obj == null) || (getClass() != obj.getClass()))
+			return false;
+		Organization other = (Organization) obj;
+		return Objects.equals(organizationID, other.organizationID)
+				&& Objects.equals(organizationName, other.organizationName);
+	}
+
+	/**
+	 * Get a builder for an {@link Organization}.
+	 *
+	 * @param organizationName the name of the organization.
+	 * @return the builder.
+	 */
+	public static Builder getBuilder(final String organizationName) {
+		return new Builder(organizationName);
+	}
+
+	/** A builder for an {@link Organization}. */
+	public static class Builder {
+
+		private final String organizationName;
+		private String organizationID = null;
+
+		private Builder(final String organizationName) {
+			this.organizationName = requireNonNull(Common.processString(organizationName), "organizationName");
```

Comment:
I would normally recommend setting a length limit on both the name and ID fields, but

1. The other provenance classes don't
2. There's a 1MB limit on the provenance data structure as a whole when converted to json